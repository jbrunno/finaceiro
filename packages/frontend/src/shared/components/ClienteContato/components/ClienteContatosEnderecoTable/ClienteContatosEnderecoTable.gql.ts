import {
  ContatoClassificacaoEnum,
  ContatoMarcadorEnum,
} from '@/shared/constants';
import { gql } from '@apollo/client';

export const CLIENTE_CONTATO_ENDERECO_FRAGMENT = {
  enderecos: gql`
    fragment CLIENTE_CONTATO_ENDERECO_FRAGMENT on ContatoEndereco {
      id
      nome
      marcador
      classificacao
      logradouro
      numero
      bairro
      cidade
      uf
      cep
      complemento
      pais {
        id
        descricao
      }
    }
  `,
};

export type ClienteContatosEndereco = {
  id: string;
  nome: string;
  marcador: ContatoMarcadorEnum | null;
  classificacao: ContatoClassificacaoEnum;
  logradouro: string;
  numero: string;
  bairro: string;
  cidade: string;
  uf: string;
  cep: string;
  complemento: string;
  pais: Pais;
};

export type Pais = {
  id: string;
  descricao: string;
};
