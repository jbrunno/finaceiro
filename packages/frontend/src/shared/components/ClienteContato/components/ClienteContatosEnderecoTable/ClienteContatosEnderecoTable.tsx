import { Fragment } from 'react';
import {
  CardText,
  CopyToClipboardButton,
  Grid,
  Table,
  TableCell,
  ToastContainer,
  Typography,
} from '@frontend/components';
import { ContatoRating } from '@/shared/components/ContatoRating/ContatoRating';
import { CONTATO_MARCADOR } from '@/shared/constants';
import {
  ClienteContatosEndereco,
  CLIENTE_CONTATO_ENDERECO_FRAGMENT,
} from './ClienteContatosEnderecoTable.gql';

type ClienteContatosEnderecoTableProps = {
  enderecos: Array<ClienteContatosEndereco>;
};

export function ClienteContatosEnderecoTable({
  enderecos,
}: ClienteContatosEnderecoTableProps) {
  return (
    <Table bodyData={enderecos.filter((_, inx) => inx < 5)}>
      {(endereco) => (
        <Fragment key={endereco.id}>
          <TableCell>
            <CardText
              label={
                endereco.marcador ? CONTATO_MARCADOR[endereco.marcador] : ''
              }
              text={endereco.nome}
            />
          </TableCell>
          <TableCell>
            <Typography variant="subtitle2" color="primary">
              {endereco.logradouro}, {endereco.numero}
            </Typography>
            <Typography variant="subtitle2" color="primary">
              {endereco.bairro}
            </Typography>
            <Typography variant="subtitle2" color="primary">
              {endereco.cidade}, {endereco.uf}
            </Typography>
            <Typography variant="subtitle2" color="primary">
              {endereco.cep}
            </Typography>
            <Typography variant="subtitle2" color="primary">
              {endereco.pais.descricao}
            </Typography>
            <Grid container direction="row" alignItems="center" gap={0.5}>
              <CopyToClipboardButton
                tooltip="Copiar endereço"
                text={`${endereco.logradouro}, ${endereco.numero} ${endereco.bairro} ${endereco.cidade} - ${endereco.uf} - ${endereco.cep} ${endereco.pais.descricao}`}
                showNotification
                textToast="Endereço copiado para a área de transferência."
              />
              <ToastContainer />
              <ContatoRating value={endereco.classificacao} />
            </Grid>
          </TableCell>
        </Fragment>
      )}
    </Table>
  );
}

ClienteContatosEnderecoTable.fragments = CLIENTE_CONTATO_ENDERECO_FRAGMENT;
