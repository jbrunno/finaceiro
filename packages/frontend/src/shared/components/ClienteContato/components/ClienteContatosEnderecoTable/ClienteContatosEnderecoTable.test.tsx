import {
  ContatoClassificacaoEnum,
  ContatoMarcadorEnum,
} from '@/shared/constants';
import { render } from '@test-utils/component-testing';
import { faker } from '@faker-js/faker';
import { ClienteContatosEnderecoTable } from './ClienteContatosEnderecoTable';

test('should render ClienteContatosEnderecoTable', () => {
  const { container } = render(
    <ClienteContatosEnderecoTable
      enderecos={[
        {
          id: '1',
          nome: 'teste',
          classificacao: ContatoClassificacaoEnum.EXCELENTE,
          marcador: ContatoMarcadorEnum.FILHO,
          cep: '69000000',
          bairro: 'Pampulha',
          cidade: 'Belo Horizonte',
          complemento: 'bl1 ap123',
          numero: '111',
          logradouro: 'Rua teste',
          uf: 'MG',
          pais: {
            id: '1',
            descricao: 'Brasil',
          },
        },
      ]}
    />,
  );
  expect(container).toMatchSnapshot();
});
