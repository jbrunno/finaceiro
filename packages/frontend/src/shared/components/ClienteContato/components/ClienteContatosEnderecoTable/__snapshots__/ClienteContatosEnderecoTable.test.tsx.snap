// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClienteContatosEnderecoTable 1`] = `
.emotion-0 {
  width: 100%;
  overflow-x: auto;
}

.emotion-1 {
  display: table;
  width: 100%;
  border-collapse: separate;
  border-spacing: 0;
}

.emotion-1 caption {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  padding: 1rem;
  color: #3B3B4199;
  text-align: left;
  caption-side: bottom;
}

.emotion-2 {
  display: table-header-group;
}

.emotion-3 {
  display: table-row-group;
}

.emotion-5 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-5.MuiTableRow-hover:hover {
  background-color: #0000000A;
}

.emotion-5.Mui-selected {
  background-color: rgba(74, 49, 255, 0.08);
}

.emotion-5.Mui-selected:hover {
  background-color: rgba(74, 49, 255, 0.12);
}

.emotion-5:hover {
  background-color: #0000000A;
}

.emotion-6 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: #3B3B41DE;
}

.emotion-9 {
  margin: 0;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.125rem;
  letter-spacing: 0.0106rem;
  font-family: Lato,sans-serif;
  color: #3B3B4199;
}

.emotion-10 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
}

.emotion-12 {
  margin: 0;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  letter-spacing: 0.0063rem;
  font-family: Lato,sans-serif;
  color: #4A31FF;
}

.emotion-17 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
}

.emotion-18 {
  padding: 0;
}

.emotion-21 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #0000008A;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #4A31FF;
  padding: 5px;
  font-size: 1.125rem;
  padding: 0;
}

.emotion-21::-moz-focus-inner {
  border-style: none;
}

.emotion-21.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-21 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-21:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-21:hover {
    background-color: transparent;
  }
}

.emotion-21:hover {
  background-color: rgba(74, 49, 255, 0.04);
}

@media (hover: none) {
  .emotion-21:hover {
    background-color: transparent;
  }
}

.emotion-21.Mui-disabled {
  background-color: transparent;
  color: #00000042;
}

.emotion-23 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 1rem;
}

.emotion-24 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-25 .Toastify__toast-container {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-25 .Toastify__toast-body {
  padding-right: 0.5rem;
}

.emotion-25 .Toastify__toast--default {
  background: #757575;
  color: #fff;
}

.emotion-25 .Toastify__toast--info {
  background: #2196F3;
  color: #fff;
}

.emotion-25 .Toastify__toast--success {
  background: #4CAF50;
  color: #fff;
}

.emotion-25 .Toastify__toast--warning {
  background: #FF9800;
  color: #fff;
}

.emotion-25 .Toastify__toast--error {
  background: #F44336;
  color: #fff;
}

.emotion-25 .Toastify__progress-bar {
  background: #fff9;
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  font-size: 1.5rem;
  color: #faaf00;
  cursor: pointer;
  text-align: left;
  -webkit-tap-highlight-color: transparent;
}

.emotion-27.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-27.Mui-focusVisible .MuiRating-iconActive {
  outline: 1px solid #999;
}

.emotion-27 .MuiRating-visuallyHidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-28 {
  cursor: inherit;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-30 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

<div>
  <div
    class="MuiTableContainer-root emotion-0"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader emotion-1"
    >
      <thead
        class="MuiTableHead-root emotion-2"
      />
      <tbody
        class="MuiTableBody-root emotion-3"
      >
        <tr
          class="MuiTableRow-root emotion-4 emotion-5"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-6"
          >
            <div
              class="MuiBox-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-caption emotion-8 emotion-9"
              >
                Filho(a)
              </span>
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-10"
              >
                teste
              </p>
            </div>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-6"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 emotion-12"
            >
              Rua teste
              , 
              111
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 emotion-12"
            >
              Pampulha
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 emotion-12"
            >
              Belo Horizonte
              , 
              MG
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 emotion-12"
            >
              69000000
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 emotion-12"
            >
              Brasil
            </h6>
            <div
              class="MuiGrid-root MuiGrid-container emotion-17"
            >
              <div
                aria-label="Copiar endereÃ§o"
                class="emotion-18 emotion-19"
                data-mui-internal-clone-element="true"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall emotion-20 emotion-21"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-22 emotion-23"
                    data-testid="FileCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4 6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root emotion-24"
                  />
                </button>
              </div>
              <div
                class="emotion-25 emotion-26"
              >
                <div
                  class="Toastify"
                />
              </div>
              <div
                aria-label="Excelente"
                class=""
                data-mui-internal-clone-element="true"
              >
                <span
                  class="MuiRating-root MuiRating-sizeMedium emotion-27"
                >
                  <label
                    class="emotion-28"
                    for=":r3:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconFilled emotion-29"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-30"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      1 Star
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":r3:"
                    name=":r2:"
                    type="radio"
                    value="1"
                  />
                  <label
                    class="emotion-28"
                    for=":r4:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconFilled emotion-29"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-30"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      2 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":r4:"
                    name=":r2:"
                    type="radio"
                    value="2"
                  />
                  <label
                    class="emotion-28"
                    for=":r5:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconFilled emotion-29"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-30"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      3 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":r5:"
                    name=":r2:"
                    type="radio"
                    value="3"
                  />
                  <label
                    class="MuiRating-label MuiRating-pristine emotion-28"
                  >
                    <input
                      class="MuiRating-visuallyHidden"
                      id=":r2:-empty"
                      name=":r2:"
                      type="radio"
                      value=""
                    />
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      Empty
                    </span>
                  </label>
                </span>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root emotion-4 emotion-5"
        />
      </tbody>
    </table>
  </div>
</div>
`;
